### PR TITLE
Create acceptance opportunity feature

### PR DESCRIPTION
**Pull request by Cxristian**
create hooks and classes for opportunity end point....

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ## CONFIGURE THE APP CLIENT.
-client_id =
-client_secret =
-user_name =
-password =
-security_token =
+client_id = 3MVG9IHf89I1t8hr66Zq39.7RKKVh5shKSUTpAo12udn.agfJ_m0Wjb8w4xOsdO_7KCD.sASrFEGhD8tw0_9H
```
Comment (alszla) [inline gradle.properties]: This file somehow managed to slip away from your surveillance, XD.

File: src/test/java/com/foundation/salesforce/hooks/OpportunityHooks.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package com.foundation.salesforce.hooks;
+
+import com.foundation.salesforce.core.api.OpportunityApi;
+import com.foundation.salesforce.entities.Context;
+import com.foundation.salesforce.entities.Opportunity;
+import cucumber.api.java.After;
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/hooks/OpportunityHooks.java:6]: Please, separate imports into groups of related librariess.

File: src/test/java/com/foundation/salesforce/hooks/OpportunityHooks.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.Map;
+
+/**
+ * OpportunityHooks class
+ *
+ * @author Cristian Lujan
+ * @version 1.0
+ */
+public class OpportunityHooks {
+
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/hooks/OpportunityHooks.java]: Please, remove this blank line.

File: src/test/java/com/foundation/salesforce/hooks/OpportunityHooks.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        Map<String,String> createNewOpportunity = new HashMap<>();
+        createNewOpportunity.put("Name", "TestApi");
+        createNewOpportunity.put("CloseDate", "2019-01-01");
+        createNewOpportunity.put("StageName", "Prospecting");
+        opportunityApi.getInstance().setContent(createNewOpportunity);
+        response = opportunityApi.createOpportunity();
+        context.setResponse(response);
+        opportunity.setId(response.jsonPath().getString("id"));
+    }
+}
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/hooks/OpportunityHooks.java]: Please, add a new blank line at the end of this file.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 5]
```diff
@@ -14,11 +14,16 @@
 
 import com.foundation.salesforce.core.api.OpportunityApi;
 import com.foundation.salesforce.core.restClient.RestClientApi;
+import com.foundation.salesforce.core.utils.ResponseValidation;
 import com.foundation.salesforce.entities.Context;
 import com.foundation.salesforce.entities.Opportunity;
+import cucumber.api.PendingException;
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java]: Please, add a seperator blank line.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 6]
```diff
@@ -14,11 +14,16 @@
...
+import cucumber.api.java.en.And;
 import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
 import cucumber.api.java.en.When;
 import io.restassured.response.Response;
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:26]: Please add a sepearator blank line.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 7]
```diff
@@ -44,22 +49,79 @@
...
-    @Given("User log in into the Opportunities page")
-    public void LogInIntoTheOpportunitiesPage() {
-        opportunityApi.getInstance().getOpportunity();
-    }
-
-
+    /**
+     * Sets a map according to input map.
+     *
+     * @param inputBody that is input data
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java]: Place a dot at the end of this line, and generally in all of the remaining parameter descriptions.

**Review by alszla — APPROVED** [review]

**Review by alszla — APPROVED** [review]

File: src/test/resources/schemas/OpportunityRetrieveSchema.json [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    "labelPlural": "string",
+    "triggerable": "boolean",
+    "undeletable": "boolean",
+    "activateable": "boolean",
+    "retrieveable": "boolean",
+    "customSetting": "boolean",
+    "replicateable": "boolean",
+    "deprecatedAndHidden": "boolean"
+  }
+}
```
Comment (TheYisusGithub) [inline src/test/resources/schemas/OpportunityRetrieveSchema.json:48]: Please put a slat of line.

File: src/test/resources/schemas/OpportunityFindSchema.json [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  "SystemModstamp": "string",
+  "HasOpenActivity": "boolean",
+  "ForecastCategory": "string",
+  "LastActivityDate": "string",
+  "LastModifiedById": "string",
+  "LastModifiedDate": "string",
+  "LastReferencedDate": "string",
+  "ForecastCategoryName": "string",
+  "HasOpportunityLineItem": "boolean"
+}
```
Comment (TheYisusGithub) [inline src/test/resources/schemas/OpportunityFindSchema.json:35]: Please put a sat of line.

**Review by titor-oopart — APPROVED** [review]

**Pull request merged**